feat(marks): add conceal_lines to nvim_buf_set_extmark() (#31324)
neovim/neovim: runtime/doc/api.txt

                    When a character is supplied it is used as |:syn-cchar|.
                    "hl_group" is used as highlight for the cchar if provided,
                    otherwise it defaults to |hl-Conceal|.
                  • conceal_lines: string which should be either empty or a

>                     single character. When provided, lines in the range are

Reviewer: Here the documentation mentions "string which should be either empty or a single character", but unlike `conceal` the API actually accepts any string without check for its content, which is a bit strange.

Author: Well the char part is as of yet unimplemented. The flag is a string to be future-proof towards that, while it is unimplemented it doesn't need to be checked in the API. But I guess there is no reason to already document the unimplemented behavior (although I do vaguely remember discussing that here or on matrix).

Reviewer 2: I don't remember the outcome; there may have been concern that people already rely on the missing check and set multiple characters, which will make implementing the missing feature a breaking change.

Maybe we can add the check already even if it's not strictly necessary yet?

Author: I think that makes sense, done. (Though the current API check for `conceal` only checks that the first char is printable anyways, not that the string is a single character. No reason to stray from that for `conceal_lines` I guess.)

                    not drawn at all; the next unconcealed line is drawn

>                     instead.

Reviewer: Mention `'conceallevel'` here?

                    When a character is supplied it is used as |:syn-cchar|.
                    "hl_group" is used as highlight for the cchar if provided,
                    otherwise it defaults to |hl-Conceal|.

>                   • conceal_line: string which should be either empty or a
>                     single character. When empty, lines in the range are not
>                     drawn at all; the next unconcealed line is drawn instead.
>                     Supplying a character is currently unimplemented.

Reviewer 2: ```suggestion
                  • conceal_line: string (empty or a
                    single character). When provided, lines in the range are not
                    drawn at all; the next unconcealed line is drawn instead.
```
(Assuming that a non-empty string is simply ignored at the moment, meaning the behavior is the same.)

Reviewer 2: > No objections, but I copied the first sentence from conceal verbatim.

That's fine then.

> If we keep that, your suggestion boils down to removing the mention of it being unimplemented?
Currently I also error when a single character is provided, could make it work like you say.

Yes, that was my suggestion.
